portal/models/organization.py

```diff
@@ -573,6 +573,15 @@ def at_or_below_ids(self, organization_id, other_organizations):
             if other_organization_id in children:
                 return True
 
+    def find_top_level_org(self, organizations):
+        orgs_list = []
+        for org in (o for o in organizations if o.id):
+            top_org_id = self.find(org.id).top_level()
+            top_org = Organization.query.filter(Organization.id == top_org_id).one_or_none()
```

Comment:
these had *better* exist - no need for the extra checks.  Also the query and code can be simplified:

`org_list.append(Organization.query.get(top_org_id))`